### PR TITLE
Feature: Extended Scope Parameters

### PR DESCRIPTION
**Pull request by fieldju**
Added ability to set extended scope parameters on the canary execution request in the manual canary executor.

image

image

image

### CODE REVIEW

File: packages/client/src/util/CanaryExecutionFactory.ts [hunk 1]
```diff
@@ -8,15 +8,15 @@ export default class CanaryExecutionFactory {
           controlScope: {
             scope: '',
             location: '',
-            step: 0,
+            step: 60,
```
Comment (melanahammel) [inline packages/client/src/util/CanaryExecutionFactory.ts]: this is a user friendly change

File: packages/client/src/validation/executionValidators.ts [hunk 2]
```diff
@@ -1,7 +1,11 @@
-import { string, object, number } from 'yup';
+import { string, object, number, array } from 'yup';
 import { CanaryExecutionRequest } from '../domain/Kayenta';
 import { ValidationResultsWrapper } from '../domain/Referee';
 
+export const DUPLICATE_SCOPE_KEYS_ERROR_MESSAGE = 'All extended query scope parameters keys must be unique.';
+export const ALL_KVS_MUST_ME_NON_EMPTY_STRINGS =
```
Comment (melanahammel) [inline packages/client/src/validation/executionValidators.ts]: small typo here, should read "BE"

File: packages/client/src/components/canary-executor/ScopeItem.tsx [hunk 3]
```diff
@@ -76,9 +113,10 @@ export default class ScopeItem extends React.Component<ScopeProps> {
         <InlineTextGroup
           id={scopeType + '-location'}
           label="Location"
+          placeHolderText="The location of your deployed service."
```
Comment (melanahammel) [inline packages/client/src/components/canary-executor/ScopeItem.tsx]: should these be flipped?
Where the description is the subtext, so users always see it,
and then the placeholder text is the example?

File: packages/client/src/util/CanaryExecutionFactory.ts (hunk 1 above)
Comment (fieldju) [reply packages/client/src/util/CanaryExecutionFactory.ts]: Yeah, I figured we could default it to what the API defaults it to.

File: packages/client/src/components/canary-executor/ScopeItem.tsx (hunk 3 above)
Comment (fieldju) [reply packages/client/src/components/canary-executor/ScopeItem.tsx]: Let's talk about this later, and we can tweak these and talk about what belongs where in regards to the placeholders, subtext and popovers (tooltips).

**Pull request merged**